ocl_web/apps/collections/views.py

```diff
@@ -601,9 +601,15 @@ def post(self, request, *args, **kwargs):
         results = result.json()
         errors = results if result.status_code == requests.codes.bad else None
 
-        if len(filter(lambda result: result['added'], results)) > 0:
+        added_result_count = len(filter(lambda result: result['added'], results))
+
+        if added_result_count > 0:
             self.add_version_warning_to_session(data, request, results)
-            self.add_mappings_to_session(request, results)
+            if self.adding_single_reference(data):
+                self.add_mappings_to_session(request, results, lambda res: 'mappings' in res['expression'])
```

Comment:
How about 
```python
request.session['add_mappings'] = [result['expression'] for result in results if 'mappings' in result['expression']]
```

instead of abstracting it to `self.add_mappings_to_session`